comictalker/talkers/comicvine.py

```diff
@@ -176,19 +169,66 @@ def __init__(
             has_nsfw=False,
             has_censored_covers=False,
         )
+        # Default settings
+        self.api_url: str = "https://comicvine.gamespot.com/api"
+        self.api_key: str = "27431e6787042105bd3e47e169a624521f89f3a4"
+        self.series_match_thresh: int = 90
+        self.remove_html_tables: bool = False
+        self.use_series_start_as_volume: bool = False
+        self.wait_for_rate_limit: bool = False
 
         # Identity name for the information source
         self.source_name: str = self.source_details.id
         self.source_name_friendly: str = self.source_details.name
 
-        self.wait_for_rate_limit: bool = wait_on_ratelimit
-        # NOTE: This was hardcoded before which is why it isn't passed in
+        tmp_url = urlsplit(self.api_url)
+
+        # joinurl only works properly if there is a trailing slash
+        if tmp_url.path and tmp_url.path[-1] != "/":
+            tmp_url = tmp_url._replace(path=tmp_url.path + "/")
+
+        self.api_url = tmp_url.geturl()
+
+        # NOTE: This was hardcoded before which is why it isn't in settings
         self.wait_for_rate_limit_time: int = 20
 
-        self.remove_html_tables: bool = remove_html_tables
-        self.use_series_start_as_volume: bool = use_series_start_as_volume
+    def comic_settings(self, parser: settngs.Manager) -> None:
+        """Register settings"""
+        parser.add_setting("--cv-use-series-start-as-volume", default=False, action=argparse.BooleanOptionalAction)
+        parser.add_setting("--cv-wait-on-ratelimit", default=False, action=argparse.BooleanOptionalAction)
+        parser.add_setting(
+            "--cv-remove-html-tables",
+            default=False,
+            action=argparse.BooleanOptionalAction,
+            help="Removes html tables instead of converting them to text.",
+        )
+        parser.add_setting(
+            "--cv-api-key",
+            help="Use the given Comic Vine API Key.",
+        )
+        parser.add_setting(
+            "--cv-url",
+            help="Use the given Comic Vine URL.",
+        )
 
-        self.series_match_thresh: int = series_match_thresh
+    def set_settings(self, settings: argparse.Namespace) -> None:
+        """Set settings."""
+        if settings.comicvine_cv_remove_html_tables:
+            self.remove_html_tables = bool(settings.comicvine_cv_remove_html_tables)
```

Comment:
https://www.youtube.com/watch?v=TJ_aGY6iyZY

Reply:
As any bool setting should have already been checked, I changed it to just equalling the value.